File: src/main/java/com/github/quiltservertools/wires/command/mute/ServerMute.java
package com.github.quiltservertools.wires.command.mute;

import java.time.Instant;

public class ServerMute {
    private boolean enabled;
    private long time;
    public ServerMute() {
        enabled = false;
        time = -1;
    }

    public void set(long time, boolean state) {
        this.time = time;
        enabled = state;
    }

    public boolean getState() {
        if (time != -1) {
            return this.enabled && time > Instant.now().getEpochSecond();
        }
        return this.enabled;
    }
}

File: src/main/kotlin/net/quiltservertools/wires/mute/ServerMute.kt
package net.quiltservertools.wires.mute

import java.time.Instant

class ServerMute {
    private var enabled = false
    private var time: Long
    operator fun set(time: Long, state: Boolean) {
        this.time = time
        enabled = state
    }

    val state: Boolean
        get() = if (time != -1L) {
            enabled && time > Instant.now().epochSecond
        } else enabled

    init {
        time = -1
    }
}